pgbar.module

```diff
@@ -55,3 +55,12 @@ function pgbar_pgbar_source_plugin_info() {
   $p['null'] = '\\Drupal\\pgbar\\Source\\StaticCount';
   return $p;
 }
+
+/**
+ * Implements hook_d7csp_hosts().
+ */
+function pgbar_d7csp_hosts() {
+  // For setting a pgbar’s initial height/width.
+  $hosts['style-src'] = "'unsafe-inline'";
+  return $hosts;
```

Comment:
This allows all any inline styles on nodes with a pgbar, or even sites with the pgbar module enabled?
Would it be an option to add the nonce to the style tag in the template?

Reply:
This is not about a `<style>` element but about `style=` attributes. The latter can’t have nonces.

We also have loads of style attributes in page content (blocks, fields, webform components). I don’t see an easy way around having this directive for sites.

Reply:
I'm just wondering if it would be more explicit to allow inline styles once for all of campaignion? On the other hand, doing it for every module makes it clear that the module requires it… :shrug:

Reply:
I ended up taking the suggestion. But still I think it’s good to list it explicitly where needed.